js/src/reader/json.ts

```diff
@@ -0,0 +1,238 @@
+// "DATA": ["49BC7D5B6C47D2","3F5FB6D9322026"]
+function readBinary(fieldObj: any, column: any) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    const offsets = readData(Int32Array, column.OFFSET);
+    // There are definitely more efficient ways to do this... but it gets the
+    // job done.
+    const joined = column.DATA.join('');
+    let data = new Uint8Array(joined.length / 2);
+    for (let i = 0; i < joined.length; i += 2) {
+        data[i >> 1] = parseInt(joined.substr(i, 2), 16);
+    }
+    return { field, fieldNode, validity, offsets, data };
+}
+
+function readNumeric<T extends TypedArray>(fieldObj: any, column: any, ArrayConstructor: TypedArrayConstructor<T>) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    const data = readData(ArrayConstructor, column.DATA);
+    return { field, fieldNode, validity, data };
+}
+
+function readInt64<T extends (Uint32Array|Int32Array)>(fieldObj: any, column: any, ArrayConstructor: TypedArrayConstructor<T>) {
+    const field = fieldFromJSON(fieldObj);
+    const fieldNode = fieldNodeFromJSON(column);
+    const validity = readValidity(column);
+    let data = new ArrayConstructor(column.DATA.length * 2);
+    for (let i = 0; i < column.DATA.length; ++i) {
+        data[2 * i  ] = column.DATA[i] >>> 0;
+        data[2 * i + 1] = Math.floor((column.DATA[i] / 0xFFFFFFFF));
+    }
+    return { field, fieldNode, validity, data };
+}
+
+function readData<T extends TypedArray>(ArrayConstructor: TypedArrayConstructor<T>, column: [number]) {
+    return new ArrayConstructor(column);
+}
+
+function readValidity(column: any) {
+    return readBoolean(column.VALIDITY);
+}
+
+function readBoolean(arr: Array<number>) {
+    let rtrn: Uint8Array = new Uint8Array(Math.ceil(arr.length / 8));
+    for (const {item, index} of arr.map((item: any, index: number) => ({item, index}))) {
+        rtrn[index / 8 | 0] |= item << (index % 8);
+    }
+    return rtrn;
+}
+
+const TYPE_LOOKUP: {[index: string]: fb.Schema.Type} = {
+    'NONE':            fb.Schema.Type.NONE,
+    'null':            fb.Schema.Type.Null,
+    'map':             fb.Schema.Type.Map,
+    'int':             fb.Schema.Type.Int,
+    'bool':            fb.Schema.Type.Bool,
+    'date':            fb.Schema.Type.Date,
+    'list':            fb.Schema.Type.List,
+    'utf8':            fb.Schema.Type.Utf8,
+    'time':            fb.Schema.Type.Time,
+    'union':           fb.Schema.Type.Union,
+    'binary':          fb.Schema.Type.Binary,
+    'decimal':         fb.Schema.Type.Decimal,
+    'struct_':         fb.Schema.Type.Struct_,
+    'floatingpoint':   fb.Schema.Type.FloatingPoint,
+    'timestamp':       fb.Schema.Type.Timestamp,
+    'fixedsizelist':   fb.Schema.Type.FixedSizeList,
+    'fixedsizebinary': fb.Schema.Type.FixedSizeBinary
+};
```

Comment:
TS enums do the identifier <-> value mappings, so I think we can avoid adding another type map here if we 
`strtoupper` on the `obj.type.name` as we read. Thoughts?

Reply:
@trxcllnt I like the idea but I don't think I can make it work since the identifiers are camel cased. If they were all lower or all upper I could just toupper or tolower `obj.type.name`, but as it stands now I don't see an easy way to make the key's case match the identifier's case for every option. Maybe I'm missing something?

I could clean it up a bit by generating the lookup programmatically:
``` typescript
const TYPE_LOOKUP: {[index: string]: Type} = {};
Object.keys(Type)   
    .filter(k => typeof Type[k as any] === "number") // filter out reverse lookups
    .forEach(k => TYPE_LOOKUP[k.toLowerCase()] = Type[k as any] as any as Type);
```